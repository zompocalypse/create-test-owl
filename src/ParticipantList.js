import React from 'react';
import './ParticipantList.css'
import Participants from './Participants'
import store from './store'

export default function ParticipantList(props) {
  return (
    <div className="participants">
      {store.participants
      .sort((a, b) => {
        if(a.inSession < b.inSession) {
          return 1;
        }
        if(a.inSession > b.inSession) {
          return -1;
        }
      })
      .map((participant) =>
        <Participants
          key={participant.id}
          name={participant.name}
          avatar={participant.avatar}
          inSession={participant.inSession}
        />
      )
    }
    </div>
  )
}