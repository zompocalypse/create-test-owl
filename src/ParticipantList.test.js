import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';
import store from './store';

describe( 'Participant component' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<ParticipantList
        {...store.participants}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})