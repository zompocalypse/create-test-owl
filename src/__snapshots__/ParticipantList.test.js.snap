// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList component renders UI as expected 1`] = `
<div
  className="participants"
>
  <div
    className="session-participant"
  >
    <img
      alt="Koren Templeton"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Koren Templeton
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Caty Flucker"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Caty Flucker
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Frank Runciman"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Frank Runciman
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Ashla Attwool"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Ashla Attwool
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Axe Kubicka"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Axe Kubicka
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
</div>
`;
